### PR TITLE
User authentication based on local data and also Social authentication

### PR DESCRIPTION
**Pull request by sudharshanreddyam**

### CODE REVIEW

File: src/routes/apiRoute.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import * as express from 'express';
+import routes from './';
```
Comment (AliakseiBychyk) [inline src/routes/apiRoute.js]: As a good practice, you can import destructured routes from `./`
`import { getProducts, getProduct, getReviews, addProduct, getUsers } from './'`
and then use them appropriately

File: src/routes/apiRoute.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import controllers from '../controllers';
```
Comment (AliakseiBychyk) [inline src/routes/apiRoute.js:3]: The same with controllers. You can import destructured controllers, that way the further code would be shorter and easier to read.

**Review by AliakseiBychyk — COMMENTED** [review]
Please provide the description how to check authentication. 
E.g. (POST `http://localhost:8080/login?password=Welcome!123&name=admin`)
It would same the time for checking.

File: src/routes/apiRoute.js (hunk 1 above)
Comment (sudharshanreddyam) [reply src/routes/apiRoute.js]: Agreed. Thanks for the suggestion. Updated the code.

File: src/routes/apiRoute.js (hunk 2 above)
Comment (sudharshanreddyam) [reply src/routes/apiRoute.js:3]: For routes, I have made the index file as json but I am not sure how to do for controllers.

**Review by AliakseiBychyk — APPROVED** [review]

**Pull request merged**